configure.ac

```diff
@@ -1254,16 +1254,8 @@ fi
 # HWLOC
 # ----------------------------------------------------------------------------
 # Allow the user to override the hwloc location (from embedded to user
-# path)
-# HWLOC
-AC_ARG_WITH([hwloc-prefix],
-            [AS_HELP_STRING([[--with-hwloc-prefix[=DIR]]],
-                            [use the HWLOC library installed in DIR,
-                             rather than the one included in src/hwloc.  Pass
-                             "embedded" to force usage of the HWLOC source
-                             distributed with MPICH.])],
-            [],dnl action-if-given
-            [with_hwloc_prefix=embedded]) dnl action-if-not-given
+# or system path)
+PAC_CHECK_PREFIX(hwloc)
```

Comment:
`PAC_CHECK_PREFIX` has the same functionality as `PAC_SET_HEADER_LIB_PATH` except the former adds `-prefix` to the option name, right? We used the latter for `libfabric` and `ucx`. I think at least we should be consistent and use a single style. 

reference: https://github.com/pmodels/mpich/pull/4118/files

Reply:
True. I was at least trying to be consistent with Hydra (https://github.com/pmodels/mpich/blob/c29531ab34c1d0e8029f2a834a52a92d67024862/src/pm/hydra/configure.ac#L421) :smile:, which had hwloc integration before MPICH. I think we can safely change both of them, though.